games/survivor/systems/modify_health.cairo

```diff
@@ -0,0 +1,8 @@
+#[system]
+fn adjust_health(query: Query<(Health)>, value: felt) {
```

Comment:
im not sure if this system is necessary. i imagine we would want a system to execute explore?

perhaps something like:

```
fn explore(survivor_id: usize) {
    let survivor: QueryOne<(Health, Name)> = query.get(survivor_id);
    ...
}
```

which could generate an execute function that takes a user defined survivor id and takes care of checking permissions ect

Reply:
yes, we will need an explore function. I created an independent health system because it is called by a few other systems, like drinking a potion, being attacked by a beast, being attacked by obstacle, etc

The explore function roughly would:

- Check auth
- Call a Random number
- Determine what the survivor has hit (beast, obstacle, found item)
- Call the corresponding system (`beast create`, `obstacle attack`, `item create`) etc
- The corresponding system would then handle a health adjustment if needed.

Reply:
Ok got it. I don't think the health system necessarily needs to be it's own contract, it could probably be a library that is used by the other systems. By default, "systems" expose an api that is called externally

Reply:
this is a pretty common pattern to update stats on a survivor. Same will exist for XP, level up etc.

Yeah ok, this makes sense. I suppose the `set` in the component is enough of an api at this stage for this.